### PR TITLE
setup.sh, doco changes

### LINKED ISSUE #597: Reset $PATH before installing
I didn't realize until recently that Spack doesn't touch the $PATH variable when generating a build environment, so theoretically what's in the user's path can affect installations. We could set the PATH variable in each site's compilers.yaml to ensure consistency of installations. In most cases, it will probably just be '/usr/bin:/bin' or similar (Spack adds its own compiler paths, etc. onto this setting).

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
### Summary

This PR:
1. Adds a '-i' option to setup.sh which attempts to reset $PATH to the system default. I believe we should try it out for our 1.5.0 release installations, and if it goes well, add it to the documentation.
2. Updates the overview documentation to clearly spell out the main use cases and where to go in the documentation for each (as well as a note on when to use spack-stack vs. Spack). It also adds a new section that is essentially the old quick start guide. I reordered a few sections; I'm not married to the current order but it feels like the logical order as far as introducing the various use cases in the Overview section.
3. Makes a few miscellaneous documentation changes (typos, stylistic tweaks, etc.).

### Testing

- For setup.sh changes, I tested concretizing and installing packages on Orion, Gaea C4, Hera, and Acorn. On each of Orion and Acorn, I ran one UFS WM RT.
- Doco updates tested with sphinx locally.

### Applications & systems affected

None directly affected, as I intend for us to further test 'setup.sh -i' where the new functionality is off by default (and I think it's fine for it to remain that way).

### Dependencies

none

### Issue(s) addressed

Addresses #746
Fixes #597

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

File: doc/source/CreatingEnvironments.rst
Comment (AlexanderRichert-NOAA) [inline doc/source/CreatingEnvironments.rst:3]: FYI @edwardhartnett

File: doc/source/CreatingEnvironments.rst
Comment (climbfuji) [inline doc/source/CreatingEnvironments.rst:19]: The NewSiteConfig instructions don't descend into the env directory but instead use `spack env activate -p envs/<environment name>/`.

Might be better to be consistent.

File: doc/source/CreatingEnvironments.rst
Comment (climbfuji) [inline doc/source/CreatingEnvironments.rst:12]: I think in other places we've used `--recursive``, maybe better to use one or the other (I don't care which) for consistency and to avoid confusion

File: doc/source/CreatingEnvironments.rst
Comment (climbfuji) [inline doc/source/CreatingEnvironments.rst]: Again, in other places we used (I believe)
```
spack install [--verbose] [--fail-fast] 2>&1 | tee log.install
```

File: doc/source/NewSiteConfigs.rst
Comment (climbfuji) [inline doc/source/NewSiteConfigs.rst]: We should be able to use the new-ish
```
spack external find --scope system --exclude ... --exclude ...
```
syntax in the copy & paste instructions to achieve the same?

File: doc/source/Overview.rst
Comment (climbfuji) [inline doc/source/Overview.rst]: At some point we want to refer to the "Maintainers" section that contains detailed instructions for maintainers of tier-1 platforms (or advanced users who want to replicate/update a tier-1 site config).

File: setup.sh
Comment (climbfuji) [inline setup.sh:4]: Intersting approach. Let's try this out for the upcoming release. I always thought that we can trim down the instructions for the tier-1 platforms ("load these modules for building spack environments") substantially, which falls into the same category as what you are trying to do here.

File: doc/source/CreatingEnvironments.rst
Comment (AlexanderRichert-NOAA) [reply doc/source/CreatingEnvironments.rst:19]: What would you think of moving to putting log.{concretize,install} under the environment itself? That way the root dir doesn't get cluttered, especially if we install additional environments.

File: doc/source/CreatingEnvironments.rst
Comment (AlexanderRichert-NOAA) [reply doc/source/CreatingEnvironments.rst:12]: 'recursive' is I believe getting deprecated, so I've changed the other instances for consistency.

File: doc/source/CreatingEnvironments.rst
Comment (AlexanderRichert-NOAA) [reply doc/source/CreatingEnvironments.rst]: updated

File: doc/source/CreatingEnvironments.rst
Comment (climbfuji) [reply doc/source/CreatingEnvironments.rst:19]: Hmm, it's super convenient to never have to `cd` elsewhere except the spack-stack top-level directory. I also found it useful to see what kind of envs exist and what has been installed. Anyway, no strong preference, but it will require some training since I've more or less hardcoded those steps in my finger muscles (and in my scripts)!

File: doc/source/NewSiteConfigs.rst
Comment (AlexanderRichert-NOAA) [reply doc/source/NewSiteConfigs.rst]: updated

File: doc/source/CreatingEnvironments.rst
Comment (AlexanderRichert-NOAA) [reply doc/source/CreatingEnvironments.rst:19]: I have a moderate preference toward putting the logs in their corresponding env directories for organizational reasons, but no strong preference on the cd. Would you prefer, say, `... | tee envs/unified-env/log.concretize`?

Comment (climbfuji) [reply doc/source/CreatingEnvironments.rst:19]: I'll leave it up to you - whatever is easier for everyone and less typing.

File: doc/source/Overview.rst
Comment (AlexanderRichert-NOAA) [reply doc/source/Overview.rst]: Just added a reference, see what you think

Comment (climbfuji) [reply doc/source/Overview.rst]: Yes, that's great, thanks very much!

**Review by climbfuji — APPROVED** [review]

**Comment by climbfuji** [comment]
Thanks for making these updates!

**Pull request merged**
